src/spyglass/decoding/v1/waveform_features.py

```diff
@@ -0,0 +1,393 @@
+        spike_times = SpikeSortingOutput.fetch_nwb(merge_key)[0]["object_id"][
+            "spike_times"
+        ]
+
+        (
+            key["analysis_file_name"],
+            key["object_id"],
+        ) = _write_waveform_features_to_nwb(
+            nwb_file_name,
+            waveform_extractor,
+            spike_times,
+            waveform_features,
+        )
+
+        AnalysisNwbfile().add(
+            nwb_file_name,
+            key["analysis_file_name"],
+        )
+        self.insert1(key)
+
+    @staticmethod
+    def _fetch_waveform(
+        merge_key: dict, waveform_extraction_params: dict
+    ) -> si.WaveformExtractor:
+        curation_key = (SpikeSortingOutput.CurationV1 & merge_key).fetch1()
+        recording = CurationV1.get_recording(curation_key)
+        if recording.get_num_segments() > 1:
+            recording = si.concatenate_recordings([recording])
+        sorting = CurationV1.get_sorting(curation_key)
+
+        waveforms_dir = temp_dir + "/" + str(curation_key["sorting_id"])
+        os.makedirs(waveforms_dir, exist_ok=True)
+
+        return si.extract_waveforms(
+            recording=recording,
+            sorting=sorting,
+            folder=waveforms_dir,
+            overwrite=True,
+            **waveform_extraction_params,
+        )
+
+    @staticmethod
+    def _compute_waveform_features(
+        waveform_extractor: si.WaveformExtractor,
+        feature: str,
+        feature_params: dict,
+        sorter: str,
+    ) -> dict:
+        feature_func = WAVEFORM_FEATURE_FUNCTIONS[feature]
+        if sorter == "clusterless_thresholder" and feature == "amplitude":
+            feature_params["estimate_peak_time"] = False
+
+        return {
+            unit_id: feature_func(waveform_extractor, unit_id, **feature_params)
+            for unit_id in waveform_extractor.sorting.get_unit_ids()
+        }
+
+    def fetch_data(self) -> tuple[list[np.ndarray], list[np.ndarray]]:
+        """Fetches the spike times and features for each unit.
+
+        Returns
+        -------
+        spike_times : list of np.ndarray
+            List of spike times for each unit
+        features : list of np.ndarray
+            List of features for each unit
+
+        """
+        return tuple(
+            zip(
+                *list(
+                    chain(
+                        *[self._convert_data(data) for data in self.fetch_nwb()]
+                    )
+                )
+            )
+        )
+
+    @staticmethod
+    def _convert_data(nwb_data) -> list[tuple[np.ndarray, np.ndarray]]:
+        feature_df = nwb_data["object_id"]
+
+        feature_columns = [
+            column for column in feature_df.columns if column != "spike_times"
+        ]
+
+        return [
+            (
+                unit.spike_times,
+                np.concatenate(unit[feature_columns].to_numpy(), axis=1),
+            )
+            for _, unit in feature_df.iterrows()
+        ]
+
+
+def _get_peak_amplitude(
```

Comment:
I would either migrate this to a decoding util or fetch the v0 version to reduce redundancy